### PR TITLE
Create dummy key using crypto iso parsing invalid key.

### LINKED ISSUE #1238: Creating a simple server crashes in an electron app
**Current behavior**

When creating a simple server running in electron crashes the server thread. This happens too in our real application which is a frontend for an ATE.

A reproduction repo is made at 

**Describe the bug**

I created a simple OPCUA server based on the provided example and started it from the main.js of my electron app.

When starting the app we get the following error:

```
➜  electron-opcua git:(main) npm start                                                                              >>>

> electron-opcua@1.0.0 start
> electron .

App threw an error during load
Error: error:06000066:public key routines:OPENSSL_internal:DECODE_ERROR
    at createPrivateKey (node:internal/crypto/keys:619:12)
    at Object.<anonymous> (/home/pti/Projects/electron-opcua/node_modules/node-opcua-secure-channel/dist/source/message_builder.js:37:59)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
    at Module._load (node:internal/modules/cjs/loader:846:12)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/pti/Projects/electron-opcua/node_modules/node-opcua-secure-channel/dist/source/index.js:25:14)
A JavaScript error occurred in the main process
Uncaught Exception:
Error: error:06000066:public key routines:OPENSSL_internal:DECODE_ERROR
    at createPrivateKey (node:internal/crypto/keys:619:12)
    at Object.<anonymous> (/home/pti/Projects/electron-opcua/node_modules/node-opcua-secure-channel/dist/source/message_builder.js:37:59)
    at Module._compile (node:internal/modules/cjs/loader:1141:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1196:10)
    at Module.load (node:internal/modules/cjs/loader:1011:32)
    at Module._load (node:internal/modules/cjs/loader:846:12)
    at f._load (node:electron/js2c/asar_bundle:2:13330)
    at Module.require (node:internal/modules/cjs/loader:1035:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/pti/Projects/electron-opcua/node_modules/node-opcua-secure-channel/dist/source/index.js:25:14)
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[9483:0206/135159.535359:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
https://github.com/ptillemans/electron-opcua
```

on line 90 of *message-builder.ts* an invalid key is created. Since electron is compiled against BoringSSL which is stricter than OpenSSL, creating this key throws the error. 

Replacing this key with the same key reformatted by pumping it through *openssl* cli tool changed the error to  key length too short.

Replacing the key with a dummy generated 1024 bit RSA key fixes the issue and the application starts up normally.

**Step by step instruction to reproduce**

check out the repo at https://github.com/ptillemans/electron-opcua

run 

```
$ npm start
```
and the electron app will start, throw the error and hang there without a visible window till Ctrl-C is pressed in the terminal.

When _fixing_ the invalid key to a valid dummy key, the app throws no errors and shows a friendly hello message.

**Expected behavior**

**Screenshots**

**Gist**

gist:

https://github.com/ptillemans/electron-opcua

**Context**

- ( ) my request is related to node-opcua acting as an OPCUA CLIENT
- (x) my request is related to node-opcua acting as an OPCUA SERVER

- ( ) I have installed node-opcua from source ( using git clone)
- (x) I have installed node-opcua as a package ( using npm install )
- ( ) I am using an application that uses node-opcua

    - ( ) node-red
    - ( ) other : please specify

- Device: Intel based PS (various)
- OS version: various Linux

    - (x) Windows : version : Win 11 Pro 22H2
    - (x) Linux : version : Arch recent, Ubuntu-14.04, Ubuntu-22.04
    - ( ) MacOs : version : ______
    - ( ) Raspbian: version : ______
    - ( ) Other : specify : ______

- Description of the other OPCUA system I am trying to connect to:
    not applicable.

    - Name: ______
    - Version: ______
    - Manufacturer/Software vendor: ______
    - link to the manufacturer or product site: https://

- **node-opcua version:** :

  
  

```
electron-opcua@1.0.0 C:\Users\PeterTillemans\Projects\electron-opcua
+-- electron@22.2.0
`-- node-opcua@2.90.0
```

- **Node:**

  `node --version` = 19.6.0

### PR DESCRIPTION
**Pull request by ptillemans**
Fixes #1238

Since `invalidPrivateKey` is AFAICT  essentially only used to have a value of the right type and to detect that the private key has not been set by the API user, a random private key can suffice as there is an insignificant chance there will ever be a collision. Also creating a private key from value 0 is a bug magnet as was demonstrated by the Java TLS implementation quite recently.

By creating a key, the key will be created by the same library which will be using it, so the likelihood of subtle conflicts between BoringSSL and OpenSSL becomes much more remote.

### CODE REVIEW

File: packages/node-opcua-secure-channel/source/message_builder.ts [hunk 1]
```diff
@@ -87,9 +87,17 @@ export interface SecurityTokenAndDerivedKeys {
     derivedKeys: DerivedKeys | null;
 }
 
-export const invalidPrivateKey = createPrivateKey(`-----BEGIN RSA PRIVATE KEY-----
-MB0CAQACAQ8CAwEAAQIBAQIBBQIBAwIBAQIBAQIBAg==
------END RSA PRIVATE KEY-----`);
+export const { invalidPrivateKey, _} = generateKeyPairSync('rsa', {
```
Comment (EirikVea) [inline packages/node-opcua-secure-channel/source/message_builder.ts]: This should be `export const { privateKey: invalidPrivateKey } = ...`

**Review by EirikVea — CHANGES_REQUESTED** [review]

**Review by EirikVea — CHANGES_REQUESTED** [review]
Actually, looks like won't work either (or at least it won't compile). After studying the documentation for [generateKeyPairSync](https://nodejs.org/api/crypto.html#cryptogeneratekeypairsynctype-options) I see that if `publicKeyEncoding` or `privateKeyEncoding` is supplied, a pair of strings will be returned. If you just remove the encoding options you will get a pair of `KeyObject`s returned, which is in turn what the `PrivateKey` type is an alias for, as far as I can tell.

**Comment by EirikVea** [comment]
> You are right again. I now spent some time to figure out how to run the tests locally on my PC.
> 
> Typefluidity in the Javascript/Typescript ecosystem is disturbing.
> 
> I updated a fixed version. I must say that it does become more elegant over time...

Yeah, I cloned your repo myself and checked that what I suggested actually compiled upfront this time 😆 
I agree with your observations on type fluidity. At least with TypeScript we have a fighting chance of detecting these sorts of things at compile time 🤯 

Anyway, nice work!

**Comment by ptillemans** [comment]
No idea why the coverage fails.

```
/home/runner/work/node-opcua/node-opcua/node_modules/.pnpm/coveralls@3.1.1/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
(Use `node --trace-uncaught ...` to show where the exception was thrown)
make: *** [Makefile:16: coveralls] Error 1
Error: Process completed with exit code 2.
```

I have no idea how this relates to the changes in message_builder.ts.

**Comment by AndreasHeine** [comment]
> No idea why the coverage fails.
> 
> ```
> /home/runner/work/node-opcua/node-opcua/node_modules/.pnpm/coveralls@3.1.1/node_modules/coveralls/bin/coveralls.js:19
>       throw err;
>       ^
> Bad response: 422 {"message":"Couldn't find a repository matching this job.","error":true}
> (Use `node --trace-uncaught ...` to show where the exception was thrown)
> make: *** [Makefile:16: coveralls] Error 1
> Error: Process completed with exit code 2.
> ```
> 
> I have no idea how this relates to the changes in message_builder.ts. 

you need rights for that!

**Comment by ptillemans** [comment]
I rebased to current master.

**Comment by erossignon** [comment]
Fixed in commit 526843c6a26b6e3d0036e53a0f227c128daa625b by simply hardcoding a valid random private key.

**Comment by erossignon** [comment]
Fixed in commit 526843c6a26b6e3d0036e53a0f227c128daa625b by simply hardcoding a valid random private key.

**Pull request closed without merging**